realm-jni/src/util.cpp

```diff
@@ -118,8 +118,8 @@ void ThrowException(JNIEnv* env, ExceptionKind exception, const std::string& cla
             break;
 
         case Unspecified:
-            jExceptionClass = env->FindClass("java/lang/RuntimeException");
-            message = "Unspecified exception. " + classStr;
+            jExceptionClass = env->FindClass("io/realm/exceptions/RealmError");
+            message = "Unrecoverable error condition. " + classStr;
```

Comment:
Throwing an error for any kind of unknown exception seems kinda harsh. I know the user can probably not do anything about it anyway, but still. 

I would rather check the error description and if it involves mmap throw an error, otherwise an exception. This is a bit of slippery slope though.